docs/documentation/upgrading/topics/keycloak/changes-25_0_0.adoc

```diff
@@ -0,0 +1,3 @@
+= Nonce claim is only added to the ID token
+
+The nonce claim is now only added to the ID token strictly following the OpenID Connect Core 1.0 specification. The spec says that that this claim is compulsory inside the https://openid.net/specs/openid-connect-core-1_0.html#IDToken[ID token] when the same parameter was sent in the authorization request. Besides it recommends to not add it after a https://openid.net/specs/openid-connect-core-1_0.html#RefreshTokenResponse[refresh request]. Previously, the claim was set to all the tokens (Access, Refresh and ID) in all the responses (refresh included). A new `Nonce backwards compatible` mapper is also included in the software that can be assigned to client scopes to revert to the old behavior.
```

Comment:
Is it possible to also add another paragraph or sentence that people may need this mapper if they use our `keycloak.js` adapter of the older version than Keycloak 24?

As there is this related bug for keycloak.js https://github.com/keycloak/keycloak/issues/26651 , which caused that JS adapter incorrectly checked the "nonce" in the access token and which was removed in Keycloak 24. So AFAIK keycloak.js applications of older version that Keycloak 24 will still have this check and hence they won't work (unless `Nonce backwards compatible` mapper is added).

Reply:
Done! Now @andymunro will do his magic. :stuck_out_tongue_closed_eyes: